### PR TITLE
Make `cargo xtask bench` work.

### LINKED ISSUE #264: xtask prevents us running #[bench] targets?
If I try `cargo xtask bench` I am told:

```
xtask: the build system does not support the bench target
```

Is this insurmountable, or is there a way to get our existing `#[bench]` targets back up and running?

### PR DESCRIPTION
**Pull request by vext01**
CC @bjorn3.

Fixes #264

### CODE REVIEW

**Comment by vext01** [comment]
Is now a good time to revisit https://github.com/softdevteam/yk/issues/197 by the way?

File: xtask/src/main.rs [hunk 1]
```diff
@@ -89,8 +89,10 @@ fn run_action(workspace: Workspace, target: &str, extra_args: &[String]) {
                 let tracing_kind = find_tracing_kind(&rust_flags);
                 rust_flags = make_internal_rustflags(&rust_flags);
 
-                // Optimise the internal workspace.
-                cmd.arg("--release");
+                if target != "bench" {
+                    // Optimise the internal workspace.
+                    cmd.arg("--release");
+                }
```
Comment (bjorn3) [inline xtask/src/main.rs:95]: Why is this necessary? Shouldn't the internal workspace always be optimized?

Comment (vext01) [reply xtask/src/main.rs:95]:
```
$ cargo bench --release
error: Found argument '--release' which wasn't expected, or isn't valid in this context
```

I guess the implication is that benchmarks are always optimised.

Comment (bjorn3) [reply xtask/src/main.rs:95]: Can you add a comment?

Comment (ltratt) [reply xtask/src/main.rs:95]: I think we can just tweak the existing comment to something like "`--bench` implies `--release`; otherwise we need to add it"?

Comment (vext01) [reply xtask/src/main.rs:96]: Try f1eb1c9

File: xtask/src/main.rs [hunk 2]
```diff
@@ -89,8 +89,11 @@ fn run_action(workspace: Workspace, target: &str, extra_args: &[String]) {
                 let tracing_kind = find_tracing_kind(&rust_flags);
                 rust_flags = make_internal_rustflags(&rust_flags);
 
-                // Optimise the internal workspace.
-                cmd.arg("--release");
+                // Optimise the internal workspace but, not for `cargo bench` where `--release` is
+                // already implied (and explicitly passing it is illegal).
```
Comment (bjorn3) [inline xtask/src/main.rs:93]: How about

```suggestion
                // Optimise the internal workspace. Don't pass `--release` for `cargo bench` however as it is
                // already implied (and explicitly passing it is illegal).
```

**Comment by vext01** [comment]
OK to squash?

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

File: xtask/src/main.rs [hunk 3]
```diff
@@ -89,8 +89,11 @@ fn run_action(workspace: Workspace, target: &str, extra_args: &[String]) {
                 let tracing_kind = find_tracing_kind(&rust_flags);
                 rust_flags = make_internal_rustflags(&rust_flags);
 
-                // Optimise the internal workspace.
-                cmd.arg("--release");
+                // Optimise the internal workspace. Don't pass `--release` for `cargo bench` however as it is
```
Comment (vext01) [inline xtask/src/main.rs:92]: CI should have failed, as this line is too long :\

More xtask debugging...
